stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -902,3 +902,49 @@ end
 *(A::Adjoint{<:Any,<:RealHermSymComplexHerm}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Adjoint{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
+
+"""
+    symmetrize!(X::AbstractMatrix; conjugate::Bool=true)
+
+Make the matrix `X` symmetric in-place using the formula `(X + X') / 2`.
+If `conjugate` is `true`, `X` is made Hermitian rather than symmetric.
+
+See also: [`symmetrize`](@ref)
+
+!!! compat "Julia 1.3"
+    This function requires Julia 1.3 or later.
+"""
+symmetrize!(X::AbstractMatrix; conjugate::Bool=true) =
+    conjugate ? _symmetrize!(conj, real, X) : _symmetrize!(identity, identity, X)
+
+function _symmetrize!(f::Function, g::Function, X::AbstractMatrix)
+    inds = axes(X, 1)
+    inds == axes(X, 2) || throw(DimensionMismatch("matrix is not square"))
+    r = first(inds)
+    s = step(inds)
```

Comment:
Is it possible to have `s != 1`?   `a[begin:end]` only works if the step is 1, so I thought we required consecutive ranges of indices.